bin/import_ts.py

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python3
+import csv
+import os
+import re
+import subprocess
+
+from multiprocessing import Pool
+from pathlib import Path
+import random
+import progressbar
+import sox
+
+try:
+    import zipfile38 as zipfile
+except ImportError:
+    print("ERROR: This importer needs additional dependencies. To fix run:")
+    print("   python -m pip install zipfile38")
+    raise
+
+from coqui_stt_training.util.downloader import SIMPLE_BAR, maybe_download
+from coqui_stt_training.util.importers import (
+    get_counter,
+    get_imported_samples,
+    get_importers_parser,
+    get_validate_label,
+    print_import_report,
+)
+
+FIELDNAMES = ["wav_filename", "wav_filesize", "transcript"]
+SAMPLE_RATE = 16000
+MAX_SECS = 15
+ARCHIVE_NAME = "2019-04-11_fr_FR"
+ARCHIVE_DIR_NAME = "ts_" + ARCHIVE_NAME
+ARCHIVE_URL = "https://drive.infomaniak.com/app/share/327683/85ec2469-8a64-4a8a-8de0-43e262d32d38/16/download"
+
+
+def _download_and_preprocess_data(target_dir, english_compatible=False):
+    # Making path absolute
+    target_dir = os.path.abspath(target_dir)
+    # Conditionally download data
+    archive_path = maybe_download(
+        "ts_" + ARCHIVE_NAME + ".zip", target_dir, ARCHIVE_URL
+    )
+    # Conditionally extract archive data
+    _maybe_extract(target_dir, ARCHIVE_DIR_NAME, archive_path)
+    # Conditionally convert TrainingSpeech data to DeepSpeech CSVs and wav
+    _maybe_convert_sets(
+        target_dir, ARCHIVE_DIR_NAME, english_compatible=english_compatible
+    )
+
+
+def _maybe_extract(target_dir, extracted_data, archive_path):
+    # If target_dir/extracted_data does not exist, extract archive in target_dir
+    extracted_path = os.path.join(target_dir, extracted_data)
+    if not os.path.exists(extracted_path):
+        print('No directory "%s" - extracting archive...' % extracted_path)
+        if not os.path.isdir(extracted_path):
+            os.mkdir(extracted_path)
+        with zipfile.ZipFile(archive_path) as zip_f:
+            zip_f.extractall(extracted_path)
+    else:
+        print('Found directory "%s" - not extracting it from archive.' % archive_path)
+
+
+def one_sample(sample):
+    """Take an audio file, and optionally convert it to 16kHz WAV"""
+    orig_filename = sample["path"]
+    # Storing wav files next to the wav ones - just with a different suffix
+    wav_filename = os.path.splitext(orig_filename)[0] + ".converted.wav"
+    _maybe_convert_wav(orig_filename, wav_filename)
+    file_size = -1
+    frames = 0
+    if os.path.exists(wav_filename):
+        file_size = os.path.getsize(wav_filename)
+        frames = int(
+            subprocess.check_output(
+                ["soxi", "-s", wav_filename], stderr=subprocess.STDOUT
+            )
+        )
```

Comment:
If you're already spawning a SoX subprocess, then you don't have to do the whole frames to duration math, just call `sox -D` instead to get a duration directly.

Reply:
Actually, even better, since we're importing the `sox` module here, it can also provide the info: https://pysox.readthedocs.io/en/latest/example.html#file-info-example

Reply:
Soooo cool! Yes yes yes 🥰